astroquery/sdss/core.py

```diff
@@ -94,76 +103,86 @@ def query_crossid_async(self, coordinates, obj_names=None,
+        spectro : bool, optional
+            Look for spectroscopic match in addition to photometric match? If
+            True, objects will only count as a match if photometry *and*
+            spectroscopy exist. If False, will look for photometric matches
+            only.
+        region : bool, optional
+            Normally cross-id only returns the closest primary object.
+            Setting this to ``True`` will return all objects.
+        field_help: str or bool, optional
+            Field name to check whether a valid PhotoObjAll or SpecObjAll
+            field name. If `True` or it is an invalid field name all the valid
+            field names are returned as a dict.
+        get_query_payload : bool, optional
             If True, this will return the data the query would have sent out,
             but does not actually do the query.
-        data_release : int
+        data_release : int, optional
             The data release of the SDSS to use.
-        """
+        cache : bool, optional
+            If ``True`` use the request caching mechanism.
 
-        if (not isinstance(coordinates, list) and
-                not isinstance(coordinates, Column) and
-                not (isinstance(coordinates, commons.CoordClasses) and
-                     not coordinates.isscalar)):
-            coordinates = [coordinates]
+        Returns
+        -------
+        result : `~astropy.table.Table`
+            The result of the query as a `~astropy.table.Table` object.
 
-        if obj_names is None:
-            obj_names = ['obj_{0}'.format(i) for i in range(len(coordinates))]
-        elif len(obj_names) != len(coordinates):
-            raise ValueError("Number of coordinates and obj_names should "
-                             "be equal")
+        """
 
-        if isinstance(radius, u.Quantity):
+        if isinstance(radius, Angle):
             radius = radius.to(u.arcmin).value
         else:
             try:
-                float(radius)
-            except TypeError:
+                radius = Angle(radius).to(u.arcmin).value
+            except ValueError:
                 raise TypeError("radius should be either Quantity or "
                                 "convertible to float.")
+        if radius > self.MAX_CROSSID_RADIUS.value:
+            raise ValueError(f"radius must be less than {self.MAX_CROSSID_RADIUS}.")
 
-        sql_query = 'SELECT\r\n'  # Older versions expect the CRLF to be there.
-
-        if specobj_fields is None:
-            if photoobj_fields is None:
-                photoobj_fields = crossid_defs
-            photobj_fields = ['p.{0}'.format(i) for i in photoobj_fields]
-            photobj_fields.append('p.objID as obj_id')
-            specobj_fields = []
+        if (not isinstance(coordinates, list) and
+            not isinstance(coordinates, Column) and
+            not (isinstance(coordinates, commons.CoordClasses) and
+                 not coordinates.isscalar)):
```

Comment:
`astropy` has a fitting function for this: https://github.com/astropy/astropy/blob/c33e53f68388d4b4d68e85956bcfe4500a94c642/astropy/utils/misc.py#L45 The only case that needs to be handled separately is `str` because every `str` is iterable.

Reply:
I'll give that a try.

Reply:
In the interest of getting this PR merged, I'm *not* going to attempt to change this, in the interest of minimizing additional changes.